<a href="https://colab.research.google.com/github/jchaves1406/Machine-Learning/blob/main/Taller2_HPC.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Universidad Sergio Arboleda**
**Programa de ciencias de la computación e inteligencia artificial**

**Escuela de ciencias exactas e ingeniería**

**Fecha:** 12/02/2022

**Autor:** Jesús Chaves Acero

**Materia:** HPC-2

**Tema:** Introducción a la analítica de datos ejercicio I.

# Taller 2: Análisis Exploratorio de Datos



### Paso 1.  Importar las Bibliotecas Necesarias

In [37]:
# Biblioteca para la manipulación y analítica de datos
import pandas as pd
import numpy as np

### Paso 2. Dataset [Enlace](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user). 

### Paso 3. Asignar una variable llamada usuarios y usar el 'user_id' como índice

In [38]:
# Se carga el dataset y se convierte en dataset
url = '/content/DataSet.csv'
dfUsuarios = pd.read_csv(url, sep = '|', index_col="user_id")
dfUsuarios.head()

age gender  occupation zip_code
user_id                                 
1         24      M  technician    85711
2         53      F       other    94043
3         23      M      writer    32067
4         24      M  technician    43537
5         33      F       other    15213

### Paso 4. Ver las primeras 25 entradas

In [39]:
# Primeras n cantidad de ocurrencias en el dataset
dfUsuarios.head(25)

age gender     occupation zip_code
user_id                                    
1         24      M     technician    85711
2         53      F          other    94043
3         23      M         writer    32067
4         24      M     technician    43537
5         33      F          other    15213
6         42      M      executive    98101
7         57      M  administrator    91344
8         36      M  administrator    05201
9         29      M        student    01002
10        53      M         lawyer    90703
11        39      F          other    30329
12        28      F          other    06405
13        47      M       educator    29206
14        45      M      scientist    55106
15        49      F       educator    97301
16        21      M  entertainment    10309
17        30      M     programmer    06355
18        35      F          other    37212
19        40      M      librarian    02138
20        42      F      homemaker    95660
21        26      M         writer    30068
22        25      M         writer    40206
23        30      F         artist    48197
24        21      F         artist    94533
25        39      M       engineer    55107

### Paso 5. Ver las últimas 10 entradas

In [40]:
# Ultimas n cantidad de ocurrencias en el dataset
dfUsuarios.tail(10)

age gender     occupation zip_code
user_id                                    
934       61      M       engineer    22902
935       42      M         doctor    66221
936       24      M          other    32789
937       48      M       educator    98072
938       38      F     technician    55038
939       26      F        student    33319
940       32      M  administrator    02215
941       20      M        student    97229
942       48      F      librarian    78209
943       22      M        student    77841

### Paso 6. ¿Cuál es el número de observaciones en el dataset?

In [41]:
# Devuelve una tupla que representa la dimensionalidad del dataset.
dfUsuarios.shape[0]

943

El dataset tiene 943 observaciones (entradas)

El numero de observaciones del Dataset es de 943 entradas

In [42]:
# Muestra la informacion de nuestro dataset
dfUsuarios.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 943 entries, 1 to 943
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   age         943 non-null    int64 
 1   gender      943 non-null    object
 2   occupation  943 non-null    object
 3   zip_code    943 non-null    object
dtypes: int64(1), object(3)
memory usage: 36.8+ KB


### Paso 7. ¿Cuál es el número de columnas en el dataset?

In [43]:
# Devuelve una tupla que representa la dimensionalidad del dataset.
dfUsuarios.shape[1]

4

El dataset tiene 4 columnas

### Paso 8. Imprime el nombre de todas las columnas.

In [44]:
# Muestra las etiquetas de las columnas de nuestro dataset
dfUsuarios.columns

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

### Paso 9. ¿Cómo esta indexed el dataset?

In [45]:
# Muestra la indexacion del dataset
dfUsuarios.index

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

El dataset esta indexado con la etiqueta "user_id" desde 1 hasta 945 y el tipo de dato es un entero. 

### Paso 10. ¿Cúal es el tipo de dato de cada columna?

In [46]:
# Muestra la informacion de nuestro dataset y el tipo de dato de cada columna
dfUsuarios.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 943 entries, 1 to 943
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   age         943 non-null    int64 
 1   gender      943 non-null    object
 2   occupation  943 non-null    object
 3   zip_code    943 non-null    object
dtypes: int64(1), object(3)
memory usage: 36.8+ KB


### Paso 11. Imprima sólo la columna ocuppation

In [47]:
# Muestra la informacion de la columna indicada
dfUsuarios['occupation']

user_id
1         technician
2              other
3             writer
4         technician
5              other
           ...      
939          student
940    administrator
941          student
942        librarian
943          student
Name: occupation, Length: 943, dtype: object

### Paso 12. ¿Cuántas ocupaciones diferentes hay en este dataset?

In [48]:
# Devuelve valores únicos basados ​​en una tabla hash.
# Se devuelven en orden de aparición. Esto NO se ordena.
dfUsuarios['occupation'].unique()

array(['technician', 'other', 'writer', 'executive', 'administrator',
       'student', 'lawyer', 'educator', 'scientist', 'entertainment',
       'programmer', 'librarian', 'homemaker', 'artist', 'engineer',
       'marketing', 'none', 'healthcare', 'retired', 'salesman', 'doctor'],
      dtype=object)

Hay 21 ocupaciones diferentes en el dataset

### Paso 13. ¿Cuál es la ocupación más frecuente?

In [49]:
# Devuelve una serie que contiene recuentos de valores únicos.
# El objeto resultante estará en orden descendente, de modo que el primer elemento 
# sea el que aparezca con más frecuencia. Excluye valores NA por defecto
dfUsuarios['occupation'].value_counts().head()

student          196
other            105
educator          95
administrator     79
engineer          67
Name: occupation, dtype: int64

La ocupacion mas frecuente del dataset es de estudiantes

### Paso 14. Resuma el DataFrame.

In [50]:
# Muestra el calculo de algunos datos estadísticos como el percentil, la media y el estándar de los valores numéricos de la serie o el marco de datos.
dfUsuarios.describe()

age
count  943.000000
mean    34.051962
std     12.192740
min      7.000000
25%     25.000000
50%     31.000000
75%     43.000000
max     73.000000

### Paso 15. Resuma todas las columnas

In [51]:
# Muestra el calculo de algunos datos estadísticos como el percentil, la media y el estándar de los valores numéricos de la serie o el marco de datos.
# 'all' Incluye estadistica de datos mixtos (no numericos).
dfUsuarios.describe(include = 'all')

age gender occupation zip_code
count   943.000000    943        943      943
unique         NaN      2         21      795
top            NaN      M    student    55414
freq           NaN    670        196        9
mean     34.051962    NaN        NaN      NaN
std      12.192740    NaN        NaN      NaN
min       7.000000    NaN        NaN      NaN
25%      25.000000    NaN        NaN      NaN
50%      31.000000    NaN        NaN      NaN
75%      43.000000    NaN        NaN      NaN
max      73.000000    NaN        NaN      NaN

### Paso 16. Resuma sólo la columna de ocupación

In [52]:
# Muestra los datos estadisticos sobre el eje solicitado
dfUsuarios['occupation'].describe()

count         943
unique         21
top       student
freq          196
Name: occupation, dtype: object

### Paso 17. ¿Cuál es la edad media de los usuarios?

In [53]:
# Devuelve la media de los valores sobre el eje solicitado.
dfUsuarios['age'].mean()

34.05196182396607

La edad media de los usuarios es de 34 años

### Paso 18. ¿Cuál es la edad de menor ocurrencia?

In [54]:
# Devuelve una serie que contiene recuentos de valores únicos.
dfUsuarios['age'].value_counts().tail(8)

68    2
64    2
69    2
7     1
66    1
11    1
10    1
73    1
Name: age, dtype: int64

Las edades con menor ocurrencia (1) en el dataset son: 7, 66, 11, 10, 73 años respectivamente